File: src/SampleCRM.Web/Services/CategoryService.cs
using OpenRiaServices.DomainServices.Hosting;
using OpenRiaServices.DomainServices.Server;
using SampleCRM.Web.Attributes;
using SampleCRM.Web.Models;
using System.Data.Entity.Migrations;
using System.Linq;

namespace SampleCRM.Web
{
    [EnableClientAccess]
    public class CategoryService : SampleCRMService
    {
        [Query]
        public IQueryable<Category> GetCategories() => 
            _context.Categories.OrderBy(x => x.Name);

        [Delete]
        [RestrictAccessReadonlyMode]
        public void DeleteCategory(Category category)
        {
            _context.Categories.Remove(category);
            _context.SaveChanges();
        }

        [Insert]
        [RestrictAccessReadonlyMode]
        public void InsertCategory(Category category)
        {

            _context.Categories.AddOrUpdate(category);
            _context.SaveChanges();
        }

        [Update]
        [RestrictAccessReadonlyMode]
        public void UpdateCategory(Category category)
        {
            _context.Categories.AddOrUpdate(category);
            _context.SaveChanges();
        }
    }
}

File: src/SampleCRM.Web/Services/CategoryService.vb
Imports OpenRiaServices.DomainServices.Hosting
Imports OpenRiaServices.DomainServices.Server
Imports SampleCRM.Web.Attributes
Imports System.Data.Entity.Migrations
Imports System.Linq

Namespace SampleCRM.Web
    <EnableClientAccess>
    Public Class CategoryService
        Inherits SampleCRMService

        <Query>
        Public Function GetCategories() As IQueryable(Of Categories)
            Return _context.Categories
        End Function

        <Delete>
        <RestrictAccessDeveloperMode>
        Public Sub DeleteCategory(ByVal category As Categories)
            _context.Categories.Remove(category)
            _context.SaveChanges()
        End Sub

        <Insert>
        <RestrictAccessDeveloperMode>
        Public Sub InsertCategory(ByVal category As Categories)
            _context.Categories.AddOrUpdate(category)
            _context.SaveChanges()
        End Sub

        <Update>
        <RestrictAccessDeveloperMode>
        Public Sub UpdateCategory(ByVal category As Categories)
            _context.Categories.AddOrUpdate(category)
            _context.SaveChanges()
        End Sub
    End Class
End Namespace
